addons/hr/tests/test_hr_contract_versions.py

```diff
@@ -0,0 +1,652 @@
+            self.assert_contract(
+                date(2025, 3, 1),
+                None,
+                False,
+                res["expected_versions_contract"][0]
+            )
+
+    def test_1contract_1version_w_date_start_date_end(self):
+        """
+        We should retrieve the contract
+        """
+        res = self.setup_1contract_1version()
+        date_versions = ['2025-01-01', '2025-06-01', '2025-12-31']
+        for date_version in date_versions:
+            self.employee.date_version = date_version
+            self.assert_contract_versions(
+                date(2025, 5, 15),
+                date(2025, 6, 15),
+                1,
+                [res["expected_versions_contract"]]
+            )
+            self.assert_contract(
+                date(2025, 5, 15),
+                date(2025, 6, 15),
+                True,
+                res["expected_versions_contract"][-1]
+            )
+            self.assert_contract(
+                date(2025, 5, 15),
+                date(2025, 6, 15),
+                False,
+                res["expected_versions_contract"][0]
+            )
+
+    def test_1contract_1version_w_date_end(self):
+        """
+        We should retrieve the contract
+        """
+        res = self.setup_1contract_1version()
+        date_versions = ['2025-01-01', '2025-06-01', '2025-12-31']
+        for date_version in date_versions:
+            self.employee.date_version = date_version
+            self.assert_contract_versions(
+                None,
+                date(2025, 8, 31),
+                1,
+                [res["expected_versions_contract"]]
+            )
+            self.assert_contract(
+                None,
+                date(2025, 8, 31),
+                True,
+                res["expected_versions_contract"][-1]
+            )
+            self.assert_contract(
+                None,
+                date(2025, 8, 31),
+                False,
+                res["expected_versions_contract"][0]
+            )
+
+    """ Timeline for this setup
+        V  : versions
+        C  : first version of the contract
+        =  : contract
+
+                     04/01=========================07/31
+        2025|C---------V---------------V------V--------VV---------|
+           01/01     04/01           06/01  07/01  07/31;08/01
+    """
+    def setup_1contract_5version(self):
+        expected_versions_contract = self.create_contract_with_versions(
+            "2025-04-01",
+            "2025-07-31",
+            self.employee.version_id,
+            [
+                "2025-04-01",
+                "2025-06-01",
+                "2025-07-01",
+                "2025-07-31",
+            ]
+        )
+        versions_not_in_contract = self.employee.create_version({'date_version': '2025-08-01'})
+        return dict(
+            expected_versions_contract=expected_versions_contract,
+            versions_not_in_contract=versions_not_in_contract
+        )
```

Comment:
Why separate the one after the contract ? The last version (1/8/2025) should still be skipped in all cases. At least it will be more explicit when you test that you get versions[-2] instead of versions[-1].
```suggestion
    def setup_1contract_5version(self):
        return self.create_contract_with_versions(
            "2025-04-01",
            "2025-07-31",
            self.employee.version_id,
            [
                "2025-04-01",
                "2025-06-01",
                "2025-07-01",
                "2025-07-31",
                "2025-08-01",
            ]
        )
```

Reply:
But in expected_versions_contract I shouldn't have the 8/01 version because it will not be synchronized with the previous contract. So in version creation I skip date_version > contract_date_end. It will not be in the "expected" versions in contract. :man_shrugging: